File: codes/Deque/Python/DequeMain.py
import DequeModule as dm

deq = dm.Deque()

deq.DQAddFirst(3)
deq.DQAddFirst(2)
deq.DQAddFirst(1)

deq.DQAddLast(4)
deq.DQAddLast(5)
deq.DQAddLast(6)

while deq.DQIsEmpty() is False:
    print(deq.DQRemoveFirst(), end=' ')
print()

deq.DQAddFirst(3)
deq.DQAddFirst(2)
deq.DQAddFirst(1)

deq.DQAddLast(4)
deq.DQAddLast(5)
deq.DQAddLast(6)

while deq.DQIsEmpty() is False:
    print(deq.DQRemoveLast(), end=' ')
print()

File: codes/Deque/C/DequeMain.c
#include <stdio.h>
#include "Deque.h"

int main(void)
{
	Deque deq;
	DequeInit(&deq);

	DQAddFirst(&deq, 3);
	DQAddFirst(&deq, 2);
	DQAddFirst(&deq, 1);

	DQAddLast(&deq, 4);
	DQAddLast(&deq, 5);
	DQAddLast(&deq, 6);

	while (!DQIsEmpty(&deq))
		printf("%d ", DQRemoveFirst(&deq));

	printf("\n");

	DQAddFirst(&deq, 3);
	DQAddFirst(&deq, 2);
	DQAddFirst(&deq, 1);

	DQAddLast(&deq, 4);
	DQAddLast(&deq, 5);
	DQAddLast(&deq, 6);

	while (!DQIsEmpty(&deq))
		printf("%d ", DQRemoveLast(&deq));

	return 0;
}